### ISSUE TITLE
Cannot restore framework

### ISSUE THREAD

**Issue by JonnyTech** [ale5000-git/tingle#19]
Currently running CM13 with CM recovery
Installed Tingle but experienced major problem
So trying to reverse using following steps:
In recovery, mount /system
$ adb push "./output/framework.jar.backup" /system/framework/framework.jar
But looks like there is no adb support in CM recovery
No other recovery exists for my phone
How can I revert?

**Comment by moodl** [comment]
@JonnyTech Can you boot into the system and enable ADB with root?

**Comment by JonnyTech** [comment]
Yes, root is available from root in normal boot but only adb sideload from cm recovery.

Did try to replace file but failed. In the end I reflashed rom.

Would be nice if a better uninstall method is implemented.

**Comment by ale5000-git** [comment]
What do you mean that there isn't adb support in CM recovery?
Can you please run from the pc `adb root`when the phone is in the recovery and tell me what it say?

PS: There will be something better in the future for uninstall.

**Comment by JonnyTech** [comment]
Booted into recovery:

```
$ adb devices
List of devices attached
* daemon not running. starting it now on port 5037 *
* daemon started successfully *
WbSPARKPJH0025693   unauthorized

$ adb root
error: device unauthorized.
This adbd's $ADB_VENDOR_KEYS is not set; try 'adb kill-server' if that seems wrong.
Otherwise check for a confirmation dialog on your device.
```

Booting to normal mode then adb works correctly.

**Comment by georgedorn** [comment]
See my workaround here, which can be done on a phone booted up normally:  https://github.com/ale5000-git/tingle/issues/22#issuecomment-285998202
